// @adyen/api-library/lib/src/typings/checkout/configuration.js
"use strict";
/**
 * Adyen Checkout API
 * Adyen Checkout API provides a simple and flexible way to initiate and authorise online payments. You can use the same integration for payments made with cards (including 3D Secure), mobile wallets, and local payment methods (for example, iDEAL and Sofort).  This API reference provides information on available endpoints and how to interact with them. To learn more about the API, visit [Checkout documentation](https://docs.adyen.com/online-payments).  ## Authentication Each request to the Checkout API must be signed with an API key. For this, obtain an API Key from your Customer Area, as described in [How to get the API key](https://docs.adyen.com/development-resources/api-credentials#generate-api-key). Then set this key to the `X-API-Key` header value, for example:  ``` curl -H \"Content-Type: application/json\" \\ -H \"X-API-Key: Your_Checkout_API_key\" \\ ... ``` Note that when going live, you need to generate a new API Key to access the [live endpoints](https://docs.adyen.com/development-resources/live-endpoints).  ## Versioning Checkout API supports versioning of its endpoints through a version suffix in the endpoint URL. This suffix has the following format: \"vXX\", where XX is the version number.  For example: ``` https://checkout-test.adyen.com/v67/payments ```
 *
 * The version of the OpenAPI document: 67
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Configuration = void 0;
class Configuration {
    static getAttributeTypeMap() {
        return Configuration.attributeTypeMap;
    }
}
exports.Configuration = Configuration;
Configuration.discriminator = undefined;
Configuration.attributeTypeMap = [
    {
        "name": "avs",
        "baseName": "avs",
        "type": "Avs"
    },
    {
        "name": "cardHolderName",
        "baseName": "cardHolderName",
        "type": "Configuration.CardHolderNameEnum"
    },
    {
        "name": "installments",
        "baseName": "installments",
        "type": "Installments2"
    },
    {
        "name": "shopperInput",
        "baseName": "shopperInput",
        "type": "ShopperInput"
    }
];
(function (Configuration) {
    let CardHolderNameEnum;
    (function (CardHolderNameEnum) {
        CardHolderNameEnum[CardHolderNameEnum["None"] = 'NONE'] = "None";
        CardHolderNameEnum[CardHolderNameEnum["Optional"] = 'OPTIONAL'] = "Optional";
        CardHolderNameEnum[CardHolderNameEnum["Required"] = 'REQUIRED'] = "Required";
    })(CardHolderNameEnum = Configuration.CardHolderNameEnum || (Configuration.CardHolderNameEnum = {}));
})(Configuration = exports.Configuration || (exports.Configuration = {}));
